### ISSUE TITLE
Internal server error happens when it tries to load the list of polls

### ISSUE THREAD

**Issue by danielepercivaldi** [v1r0x/polls#14]
Using:

- Ubuntu 14.4 64bit
- owncloud 8.1.5
- firefox 44.0.2
- chrome 48.0.2564.116 m

Below the console log:
https://nerv.net.intre.it/core/vendor/jquery/jquery.min.js?v=e2dd59b96267903f7a35f4e5c2d57557 Failed to load resource: the server responded with a status of 412 (Precondition failed)send @ jquery.min.js?v=e2dd59b96267903f7a35f4e5c2d57557:6

Below owncloud log when error occurs:
[owncloud.txt](https://github.com/v1r0x/polls/files/163540/owncloud.txt)

**Comment by v1r0x** [comment]
Does this happen after the blank page workaround or even with a unmodified oc instance?

**Comment by danielepercivaldi** [comment]
After the blank page workaround.

**Comment by v1r0x** [comment]
Maybe something is broken. Can't see an error of the polls app. Try to revert the changes and check if it works.

**Comment by danielepercivaldi** [comment]
Hi.
Sorry, I wrong.
Internal server error occurs with and without blank page workaround (BPW).
Here the sequence of events that has given the situation:

1. installed polls 0.6.9
2. added a poll without BPW. Result: blank page.
3. applied BPW
4. added poll. Result: poll was added and few seconds after loading poll list server internal error occurs
Hope this can help you.

**Comment by v1r0x** [comment]
Could you try to revert the BPW and create a new poll and then load the poll list?

**Comment by danielepercivaldi** [comment]
Just discovered what happened: I installed your app before deleting the polls of the old one ... now with BPW it works fine.
Sorry.

**Comment by v1r0x** [comment]
Glad to hear that!

**Comment by danielepercivaldi** [comment]
Sorry to disturb you again, but the error happens again.
When I backed to the old version after "internal server error" I found a poll with creation and expiration date on 1/1/1970 so I thought thet it was an old poll post database conversion for the new app.
It seems it works only once in order to add one poll and then ... internal server error, w/o BPW.

**Comment by v1r0x** [comment]
Could you set your oc debug level in `config.php` to `0` and create a new poll and post the oc log here, please.

**Comment by danielepercivaldi** [comment]
Log with BPW:

[owncloud.log.txt](https://github.com/v1r0x/polls/files/164095/owncloud.log.txt)

**Comment by danielepercivaldi** [comment]
Log without BPW:

[owncloud.log.2.txt](https://github.com/v1r0x/polls/files/164100/owncloud.log.2.txt)

**Comment by v1r0x** [comment]
Ok, maybe I found the problem. Could you check in the database (`SELECT hash FROM oc_polls_events`) or somewhere else if there is a `/` in the poll url? (Something like `https://yourdomain.tld/owncloud/index.php/apps/polls/goto/<hash>`)

**Comment by danielepercivaldi** [comment]
Well, I checked all the tables, but you must know that now the app does not add any poll, so the list is empty, BUT, checking all the tables, I found all other info I entered (texts and dts, for instance).
I don't know, stranger and stranger.

**Comment by v1r0x** [comment]
I'm not able to figure out whats the problem. The two logfiles have completely different errors. You could try to delete the app, reinstall the current master branch and increase the version in `polls/appinfo/info.xml` to something like 0.6.9.1 and force a db update. I think something in your DB is messed up with the old and new version.

**Comment by danielepercivaldi** [comment]
Done. Database updated.
Nothing: no more internal server error but no poll added ...

**Comment by v1r0x** [comment]
Any new errors in the js console or oc log?

**Comment by danielepercivaldi** [comment]
Nothing in console.
In log two errors about pollid. Are those yours?

![cattura](https://cloud.githubusercontent.com/assets/8720648/13648950/9b555718-e63b-11e5-8b7c-fb8aa69a3bad.JPG)

**Comment by v1r0x** [comment]
Yes, looks like a typo. Could you please test the branch with the fix from #17 ?
Thanks for your comments and all the testing so far!

**Comment by danielepercivaldi** [comment]
Sorry, not yet any poll in list after adding it.
All oc_polls tables are empty but:

- oc_polls_txts

"id","poll_id","text"
"1","","qwerty3"
"2","","qwerty1"
"3","","qwerty2"
"4","","test13"
"5","","test11"
"6","","test13"
"7","","test11"
"8","-1","test11"
"9","-1","test13"
"10","-1","test14"
"11","-1","test1"
"12","-1","test1"
"13","-1","test23"
"14","-1","test21"

- oc_polls_dts

"id","dt","poll_id"
"2","0000-00-00 00:00:00",""
"3","0000-00-00 00:00:00",""

**Comment by v1r0x** [comment]
Found another typo. Please try again. :)

**Comment by danielepercivaldi** [comment]
Good. Now it works once. Then I removed the first added poll, added a new one and finally ... internal server error.

**Comment by v1r0x** [comment]
Looks like the hash generator adds `\` and `/` to it, which is prohibited. I'll try to fix this later or tomorrow. As a workaround you could try to replace all `\` and `/` in the hashs in oc_polls_events.

**Comment by danielepercivaldi** [comment]
You are rigth.
This explains why sometimes internal server error occurs and sometimes not: probability of hash generation with / and/or \ .
Ok. Let's go on tomorrow.

Goodnight!

**Comment by v1r0x** [comment]
Should be fixed now. Please check out latest patch in the #17 branch and check if new polls work for you (remember to remove the old ones with / and \)

**Comment by v1r0x** [comment]
Couldn't reproduce it, but changed something. Maybe this works? Please test again the updated branch.

**Comment by danielepercivaldi** [comment]
Sorry ...

![cattura](https://cloud.githubusercontent.com/assets/8720648/13716439/550497d2-e7db-11e5-8e2c-c54dc54a082a.JPG)
![cattura](https://cloud.githubusercontent.com/assets/8720648/13716458/7a264f1a-e7db-11e5-8421-ca601caf6f16.JPG)

J console:
POST https://nerv.net.intre.it/index.php/apps/polls/insert 500 (Internal Server Error)

**Comment by v1r0x** [comment]
This is weird. Could you please checkout the branch again and test the new commit?

**Comment by danielepercivaldi** [comment]
![cattura](https://cloud.githubusercontent.com/assets/8720648/13724537/209d6a10-e889-11e5-9a4d-5efbe146c618.JPG)

**Comment by v1r0x** [comment]
Ok, now please test again.

**Comment by danielepercivaldi** [comment]
![cattura](https://cloud.githubusercontent.com/assets/8720648/13726112/4a78c3a4-e8b9-11e5-86d5-efb40e63c16a.JPG)

**Comment by v1r0x** [comment]
Ok, let's try this one more time!
Please make sure you are on the `fix-pollid-typo` branch and the latest commit is `ea29b677dfab07bd570ad8aac820b57821347c60`.

**Comment by danielepercivaldi** [comment]
Just to be sure, this is what I do:

cd /var/www/owncloud/apps
git clone https://github.com/v1r0x/polls.git
cd polls
git checkout fix-pollid-typo

Right? Anything else? Or is anything wrong?

**Comment by danielepercivaldi** [comment]
No.
![cattura](https://cloud.githubusercontent.com/assets/8720648/13731702/1d469ebc-e970-11e5-9217-c03c35950601.JPG)

**Comment by v1r0x** [comment]
Exactly. Should work now. @LukasReschke reminded me how dumb I am ;)

**Comment by v1r0x** [comment]
I added a new commit which should work.

**Comment by danielepercivaldi** [comment]
It seems ok now. Well, no internal server error when adding, but when voting now. And other errors.

**Comment by v1r0x** [comment]
Let's merge this one and please open a new issue with your problem and a log. Thanks for all the help!
